backstory/src/main/java/com/google/sps/story/PromptManagerAPIsClient.java

```diff
@@ -25,33 +25,29 @@
  * Facilitates API calls through a single object.
  */
 public class PromptManagerAPIsClient {
-  /** Client object for word classification */
-  private NLServiceClient wordClassifier;
   /** Client object for word fetching */
   private DatamuseRequestClient wordFetcher;
 
   /**
-   * Initialize API objects.
-   *
-   * @throws IOException Exception for network problem.
+   * Initialize Datamuse API object.
    */
   public PromptManagerAPIsClient() throws IOException {
-    try {
-      // Instantiate API objects
-      wordClassifier = new NLServiceClient();
-      wordFetcher = new DatamuseRequestClient();
-    } catch (IOException ioException) {
-      throw ioException;
-    }
+    wordFetcher = new DatamuseRequestClient();
   }
 
   /**
    * Classifies an input of strings based on their part of speech.
    *
    * @param words A list of Strings containing keywords for prompts.
    * @return A mapping of WordTypes to given words.
+   * @throws IOException Exception for network problem.
    */
-  public Map<WordType, List<String>> groupByWordType(List<String> words) {
+  public Map<WordType, List<String>> groupByWordType(List<String> words) throws IOException {
+    NLServiceClient wordClassifier = new NLServiceClient();
+
+    Map<WordType, List<String>> groupings = wordClassifier.groupByWordType(words);
+    wordClassifier.close();
```

Comment:
What does it mean to close the client? It looks like the client is used again after closing it, which is somewhat unintuitive.

Reply:
You're totally right. The intent is actually that the client is not used again. That was a typo by me. I changed it. Closing the client really means closing the connection with the API (the LanguageServiceClient created in the constructor).

Reply:
Done